github/plugin.py

```diff
@@ -735,3 +787,84 @@ def invite_user_to_gh_org_team(self, github_id, team_id):
+
+        Command is as follows: !listprs <organization> <repo> <state>
+        :param data:
+        :return:
+        """
+        try:
+            parser = argparse.ArgumentParser()
+            parser.add_argument('org', type=str)
+            parser.add_argument('repo', type=str)
+            parser.add_argument('state', type=str)
+
+            args, unknown = parser.parse_known_args(args=preformat_args(data["text"]))
+            if len(unknown) > 0:
+                raise SystemExit()
+
+            args = vars(args)
+
+            # Check that we can use this org:
+            real_org = self.org_lookup[args["org"]][0]
+            reponame = extract_repo_name(args["repo"])
+
+            #Check if the sent state is permitted
+            state = args["state"]
+            if state not in self.commands["!ListPRs"]["permitted_states"]:
+                raise KeyError("PRStates")
+
+        except KeyError as ke:
+            if "PRStates" in str(ke):
+                s_str = " or ".join(["`{perm_state}`".format(perm_state=perm_state)
+                                     for perm_state in self.commands["!ListPRs"]["permitted_states"]])
+                send_error(data["channel"], '@{}: Invalid state sent in.  States must be {perm_states}.'
+                           .format(user_data["name"], perm_states=s_str), markdown=True)
+            else:
+                send_error(data["channel"], '@{}: Invalid orgname sent in.  Run `!ListOrgs` to see the valid orgs.'
+                           .format(user_data["name"]), markdown=True)
+            return
+
+        except SystemExit as _:
+            send_info(data["channel"], "@{}: `!ListPRs` usage is:\n```!ListPRs <OrgThatHasRepo> "
+                                       "<Repo> <State>```\n"
+                                       "No special characters or spaces in the variables. \n"
+                                       "Run `!ListOrgs` to see the list of GitHub Organizations that I manage. "
+                      .format(user_data["name"]), markdown=True)
+            return
+
+        # Auth?
+        if self.commands["!ListPRs"].get("auth"):
+            if not self.commands["!ListPRs"]["auth"]["plugin"].authenticate(
+                    data, user_data, **self.commands["!ListPRs"]["auth"]["kwargs"]):
+                return
+
+        # Output that we are doing work:
+        send_info(data["channel"], "@{}: Working, Please wait...".format(user_data["name"]))
+
+        # Check that the repo exists:
+        repo_data = self.check_gh_for_existing_repo(reponame, real_org)
+        if not (repo_data):
+            send_error(data["channel"],
+                       "@{}: This repository does not exist in {}.".format(user_data["name"], real_org))
+            return
+
+        # Grab all PRs [All states]
+        pull_requests = self.get_repo_prs(data, user_data, reponame, real_org, state)
+        if not (pull_requests):
+            if isinstance(pull_requests, list):
```

Comment:
@mikegrima Yep, just because empty lists match the condition `if not (pull_requests):`, and since we might return a boolean from `get_repo_prs` as well to stop the execution of the remaining section.

If we return before continuing the execution, we won't be able to reply with the message info.

For `send_info`, it's already there 🚶 ...